components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/authz/OAuthAuthzReqMessageContext.java

```diff
@@ -74,6 +77,20 @@ public void setApprovedScope(String[] approvedScope) {
         this.approvedScope = approvedScope;
     }
 
+    /**
+     * @return user requested scope list
+     */
+    public String[] getRequestedScopes() {
+        return requestedScopes;
```

Comment:
We need to return a copy here.
@shashimalcse 

Otherwise the requested scopes can be tampered to bypass the validation that checks if approved scopes == requested scopes